src/averageAges.js

```diff
@@ -55,7 +73,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const peopleCopy = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : [...people];
+
+  const ageDifferenceArr = peopleCopy
+    .map(person => {
+      const mother = people.find(per => per.name === person.mother);
+
+      const difference = mother
+        ? person.born - mother.born
+        : false;
+
+      return difference;
+    })
+    .filter(age => age);
```

Comment:
Why do you need to filter it?

Reply:
But why do you need to map the array with a mother object and possibly a false value,
if you could create an array with mothers with one filter?